shared/tests/utility/math/TestCoordinates.cpp

```diff
@@ -0,0 +1,134 @@
+/*
+ * This file is part of the NUbots Codebase.
+ *
+ * The NUbots Codebase is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The NUbots Codebase is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with the NUbots Codebase.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright 2021 NUbots <nubots@nubots.net>
+ */
+
+#include <Eigen/Core>
+#include <array>
+#include <catch.hpp>
+#include <cfloat>
+#include <cmath>
+#include <fmt/format.h>
+#include <iomanip>
+#include <utility>
+#include <yaml-cpp/yaml.h>
+
+#include "utility/math/coordinates.hpp"
+#include "utility/support/yaml_expression.hpp"
+
+using utility::math::coordinates::cartesianToSpherical;
+using utility::math::coordinates::sphericalToCartesian;
+using utility::support::Expression;
+using utility::support::resolve_expression;
+
+
+/**
+ * @author Liam Craft
+ */
+
+static constexpr double ERROR_THRESHOLD = 1e-6;
+
+/** Test Data */
+static const YAML::Node test_values = YAML::LoadFile("tests/CoordinateTests.yaml");
+
+// cartesian to spherical
+SCENARIO("Cartesian coordinates can be converted to spherical coordinates", "[utility][math][coordinates]") {
+    GIVEN("A set of cartesian coordinates and a set of expected values") {
+        // cart input
+        static const std::array<Eigen::Vector3d, 200> cart_coords =
```

Comment:
Might be good to specify the number of points in the yaml file instead of hardcoding here? or use an `std::vector` if possible?

Reply:
Just for my curiosity - what's the advantage of that?

Reply:
new set of data?

Reply:
IIRC, the blame is on me for using an array here. I guess I figured that if you know the size at compile time and it doesn't change, it should be an array, but (at the risk of speaking for Liam here) I would be totally fine with vector here  if you think that's better (I don't think you can use a std::array with a dynamic size/a size only found at runtime).

Reply:
Doesn't bother me either way. Might be cleaner with a vector